app/controllers/api/v1/vacations_controller.rb

```diff
@@ -7,6 +7,10 @@ def index
     respond_with @vacations
   end
 
+  def vacation_reasons
```

Comment:
I prefer REST approach in actions. REST approach says that you have CRUD type of resources.
In your case, it is better to have `resources :type` in the `vacation` namespaces.

About field type. In the model, we have a reason, on front end we have `vacationTypes`. It can be confusing when we have much more code.

I like `type`. It literally says what is it.

Reply:
Also, it would be nice to add some tests.